RepositoryBase.js

```diff
@@ -263,24 +263,32 @@ class RepositoryBase extends PObject {
    *
    * @param {string} id The ID of the element we want to update.
    * @param {Array} data The data we need to store.
+   * @param {Object|null} [transaction=null] The transaction we want to use.
    * @throws RepositoryException
    *
    * @return {Object}
    */
-  update(id, data) {
-    return this.updateByCriteria(new IDCriteria(id), data);
+  update(id, data, transaction = null) {
+    const criteria = new IDCriteria(id);
+    criteria.transaction = transaction;
```

Comment:
Why don't you pass the transaction to the constructor?

Reply:
To allow same repository instance to execute multiple, independent, operations. The transaction should not be a property of the repository, it should only be a parameter to the operations executed by the repository.

Reply:
Added transaction through criteria constructor